bundler/lib/bundler/injector.rb

```diff
@@ -86,7 +86,7 @@ def conservative_version(spec)
       segments = version.segments
       seg_end_index = version >= Gem::Version.new("1.0") ? 1 : 2
 
-      prerelease_suffix = version.to_s.gsub(version.release.to_s, "") if version.prerelease?
+      prerelease_suffix = version.to_s.sub(version.release.to_s, "") if version.prerelease?
```

Comment:
I think this is going to be a change in behavior -- worth adding a test for when the release string is duplicated

Reply:
`1.0.0.pre.1.0.0` would be an example, I think

Reply:
But that would he incorrect to replace both.

Reply:
yes, but im saying we should test it since you're changing the code around it

Reply:
Got it. Agree.

Reply:
Ok, added a test that fails with the old version with `gsub` but not with `sub`:
```diff
-gem "lemur", "~> 3.1.pre.2023.1.1"
+gem "lemur", "~> 3.1.pre.202"
```

Reply:
another idea: `prerelease_suffix = version.to_s.delete_prefix(version.release.to_s) if version.prerelease?` or even `version._split_segments[1].join(".")`